src/python/pants/jvm/strip_jar/strip_jar.py

```diff
@@ -0,0 +1,201 @@
+
+_STRIP_JAR_BASENAME = "StripJar.java"
+_OUTPUT_PATH = "__stripped_jars"
+
+
+class StripJarToolLockfileSentinel(GenerateToolLockfileSentinel):
+    resolve_name = "strip-jar"
+
+
+@dataclass(frozen=True)
+class StripJarRequest:
+    digest: Digest
+    filenames: tuple[str, ...]
+
+
+@dataclass(frozen=True)
+class FallibleStripJarResult:
+    process_result: FallibleProcessResult
+
+
+@dataclass(frozen=True)
+class StripJarCompiledClassfiles:
+    digest: Digest
+
+
+@rule(level=LogLevel.DEBUG)
+async def strip_jar(
+    processor_classfiles: StripJarCompiledClassfiles, jdk: InternalJdk, request: StripJarRequest
+) -> FallibleStripJarResult:
+
+    filenames = list(request.filenames)
+
+    input_path = "__jars_to_strip"
+    toolcp_relpath = "__toolcp"
+    processorcp_relpath = "__processorcp"
+
+    lockfile_request = await Get(GenerateJvmLockfileFromTool, StripJarToolLockfileSentinel())
+
+    tool_classpath, prefixed_jars_digest = await MultiGet(
+        Get(
+            ToolClasspath,
+            ToolClasspathRequest(lockfile=lockfile_request),
+        ),
+        Get(Digest, AddPrefix(request.digest, input_path)),
+    )
+
+    extra_immutable_input_digests = {
+        toolcp_relpath: tool_classpath.digest,
+        processorcp_relpath: processor_classfiles.digest,
+    }
+
+    process_result = await Get(
+        FallibleProcessResult,
+        JvmProcess(
+            jdk=jdk,
+            classpath_entries=[
+                *tool_classpath.classpath_entries(toolcp_relpath),
+                processorcp_relpath,
+            ],
+            argv=["org.pantsbuild.stripjar.StripJar", input_path, _OUTPUT_PATH, *filenames],
+            input_digest=prefixed_jars_digest,
+            extra_immutable_input_digests=extra_immutable_input_digests,
+            output_directories=(_OUTPUT_PATH,),
+            extra_nailgun_keys=extra_immutable_input_digests,
+            description=f"Stripping jar {filenames[0]}",
+            level=LogLevel.DEBUG,
+        ),
+    )
+
+    return FallibleStripJarResult(process_result=process_result)
+
+
+@rule(level=LogLevel.DEBUG)
+async def resolve_fallible_result_to_strip_jar(
+    fallible_result: FallibleStripJarResult,
+    process_cleanup: ProcessCleanupOption,
+) -> Digest:
+    if fallible_result.process_result.exit_code == 0:
+        digest = await Get(
+            Digest, RemovePrefix(fallible_result.process_result.output_digest, _OUTPUT_PATH)
+        )
+        return digest
+    raise ProcessExecutionFailure(
+        fallible_result.process_result.exit_code,
+        fallible_result.process_result.stdout,
+        fallible_result.process_result.stderr,
+        "Strip jar failed.",
+        process_cleanup=process_cleanup.val,
+    )
+
+
+def _load_strip_jar_source() -> bytes:
+    return pkg_resources.resource_string(__name__, _STRIP_JAR_BASENAME)
+
+
+@rule
+async def build_processors(jdk: InternalJdk) -> StripJarCompiledClassfiles:
```

Comment:
It would be good to reference https://github.com/pantsbuild/pants/issues/13879 here (and wherever this `@rule` was copied from).

Reply:
fixed.